packages/odyssey-react-mui/src/MenuButton.tsx

```diff
@@ -93,25 +81,28 @@ export type MenuButtonProps = {
    * The tooltip text for the Button if it's icon-only
    */
   tooltipText?: string;
-} & (
-  | {
-      ariaLabel?: string;
-      ariaLabelledBy?: string;
-      buttonLabel: string;
-    }
-  | {
-      ariaLabel: string;
-      ariaLabelledBy?: string;
-      buttonLabel?: undefined | "";
-    }
-  | {
-      ariaLabel?: string;
-      ariaLabelledBy: string;
-      buttonLabel?: undefined | "";
-    }
-) &
-  Pick<FieldComponentProps, "isDisabled"> &
-  HtmlProps;
+} & Pick<
+  HtmlProps,
+  "ariaDescribedBy" | "ariaLabel" | "ariaLabelledBy" | "testId" | "translate"
+> &
+  (
+    | {
+        ariaLabel?: string;
+        ariaLabelledBy?: string;
+        buttonLabel: string;
+      }
+    | {
+        ariaLabel: string;
+        ariaLabelledBy?: string;
+        buttonLabel?: undefined | "";
+      }
+    | {
+        ariaLabel?: string;
+        ariaLabelledBy: string;
+        buttonLabel?: undefined | "";
```

Comment:
Just wanted to check in about this. Have you tested these 3 different use-cases in code somewhere (storybook or somewhere else) to make sure TS can figure out which of these is true? Reason I ask is for the following. If I set the following props:
`<MenuButton ariaLabel="foo" ariaLabelledBy="bar" buttonLabel="">...</MenuButton>`
How does TS know which case this is? Or does it care?

Reply:
@chrispulsinelli-okta Made a change here if you want to take another look. The intention here is that if `buttonLabel` is `undefined | ""` either `ariaLabel` OR `ariaLabelledby` is required. It works but, the TS error isn't quite as clear as I'd like. Picks whichever condition is last in the order and says that one is required. If that makes sense...